Minio/DataModel/CopyConditions.cs

```diff
@@ -84,13 +100,37 @@ public void SetMatchETag(string etag)
          * @throws InvalidArgumentException
          *           When etag is null
          */
-        public void SetMatchETagNone(string etag)  
+        public void SetMatchETagNone(string etag)
         {
             if (etag == null)
             {
                 throw new ArgumentException("ETag cannot be empty");
             }
-            copyConditions["x-amz-copy-source-if-none-match"] =  etag;
+            copyConditions.Add("x-amz-copy-source-if-none-match", etag);
+        }
+
+        /**
+       * Set Byte Range condition, copy object which falls within the 
+       * start and end byte range specified by user
+       *
+       * @throws InvalidArgumentException
+       *           When firstByte is null or lastByte is null
+       */
+        public void SetByteRange(long firstByte, long lastByte)
+        {
+            if ((firstByte < 0) || (lastByte < firstByte))
+                throw new ArgumentException("Range start less than zero or range end less than range start");
+            if ((lastByte - firstByte + 1) < 1)
+                throw new ArgumentException("Offset must refer to a non zero range length.");
```

Comment:
This condition is not required - as it is satisfied by the previous if condition. Size will be always non-zero here.